python/social_bot/envs/embodied_teacher.py

```diff
@@ -0,0 +1,265 @@
+                this is set to 0. if 1:1 real time is prefered(like playing or recording video),
+                this should be set to 1.0/world_time_precision.
+            port: Gazebo port, need to specify when run multiple environment in parallel
+            action_cost (float): Add an extra action cost to reward, which helps to train
+                an energy/forces efficency policy or reduce unnecessary movements
+            resized_image_size (None|tuple): If None, use the original image size
+                from the camera. Otherwise, the original image will be resized
+                to (width, height)
+            vocab_sequence_length (int): the length of encoded sequence
+            is_training_for_teacher_policy (bool): if true this env is acting like a normal
+                gym environment without teacher, used for training the teacher policy
+            demo_by_human (bool): demo by human or by existing teacher policy
+        """
+        self._is_training_for_teacher_policy = is_training_for_teacher_policy
+        self._demo_by_human = demo_by_human and (
+            not is_training_for_teacher_policy)
+        if self._is_training_for_teacher_policy:
+            self.step = self.step
+        elif self._demo_by_human:
+            from social_bot.keybo_control import KeyboardControl
+            self.step = self._step_with_human_demo
+            self._keybo = KeyboardControl()
+            real_time_update_rate = 500  # run "gz physics -u" to override
+        else:  # demo by teacher policy
+            self.step = self._step_with_teacher_policy
+
+        super().__init__(
+            agent_type=agent_type,
+            world_name="play_ground.world",
+            tasks=tasks,
+            with_language=with_language,
+            use_image_observation=use_image_observation,
+            image_with_internal_states=image_with_internal_states,
+            world_time_precision=world_time_precision,
+            step_time=step_time,
+            real_time_update_rate=real_time_update_rate,
+            port=port,
+            action_cost=action_cost,
+            resized_image_size=resized_image_size,
+            vocab_sequence_length=vocab_sequence_length)
+            
+        if not self._is_training_for_teacher_policy:
+            # insert teacher model
+            self.insert_model(
+                model=agent_type, name="teacher", pose="0 -2 0 0 0 0")
+            # set up teacher joints
+            self._teacher_joints = []
+            for joint in self._agent_joints:
+                self._teacher_joints.append("teacher::" + joint)
+            self._teacher_embodied = self._world.get_agent('teacher')
+
+    def reset(self):
+        if self._demo_by_human:
+            self._keybo.reset()
+        obs = super().reset()
+        if self._demo_by_human or self._is_training_for_teacher_policy:
+            return obs
+        else:
+            return obs, obs
```

Comment:
Better to use Dict and also need to set the correct observation space and action space in __init__

Reply:
Done. Changed to OrderedDict similar to Haichao's experiment here https://github.com/Haichao-Zhang/SocialRobot/commit/f918287043804f6c7ad9ce5804992ccd001d1dcc